### ISSUE TITLE
Intermediate code generation with generics... type Object does not take parameters

### ISSUE THREAD

**Issue by brianloss** [stoicflame/enunciate#972]
I'm using enunciate to generate API documentation, and one of our classes has a generic type that extends Comparable. Enunciate is generating invalid intermediate code that does not compile. I've been able to modify the sample app to reproduce the issue. The only change I made to the source code is to modify the Name class such that the class definition is:
```java
public class Name<S extends Comparable<S>> {
```
Then running `mvn package` on the sample produces the following error:
```bash
[INFO] --- enunciate-maven-plugin:2.11.1:assemble (default) @ enunciate-sample ---
[INFO] [ENUNCIATE] Using enunciate configuration at /home/myuser/Projects/enunciate-sample/enunciate.xml
[INFO] Reflections took 667 ms to scan 50 urls, producing 85 keys and 85 values 
/home/myuser/Projects/enunciate-sample/target/enunciate/java-json-client/src/com/ifyouwannabecool/domain/persona/Name.java:17: error: type Object does not take parameters
public class Name<S extends java.lang.Object<S>> implements java.io.Serializable {
                                            ^
1 error
```

**Comment by brianloss** [comment]
I should add that version 2.10.0 does not fail. Perhaps the fix for issue #775 caused my problem?

**Comment by stoicflame** [comment]
[Enunciate 2.12](https://github.com/stoicflame/enunciate/releases/tag/v2.12.0) has been released.
